zerver/tests/test_auth_backends.py

```diff
@@ -1054,26 +1062,56 @@ def stage_two_of_registration(self, result: HttpResponse, realm: Realm, subdomai
         else:
             self.assertIn('do_confirm/' + confirmation_key, result.url)
             do_confirm_url = result.url
+
+        if user_avatar_url:
+            # If there is an avatar_url, then there must be a user_avatar_url
+            # on the preregistration user.
+            prereg_user = confirmation.content_object
+            self.assertEqual(prereg_user.user_avatar_url, user_avatar_url)
+
         result = self.client_get(do_confirm_url, name = name)
         self.assert_in_response('action="/accounts/register/"', result)
         data = {"from_confirmation": "1",
                 "key": confirmation_key}
+
         result = self.client_post('/accounts/register/', data)
         if not skip_registration_form:
             self.assert_in_response("We just need you to do one last thing", result)
 
             # Verify that the user is asked for name but not password
             self.assert_not_in_success_response(['id_password'], result)
             self.assert_in_success_response(['id_full_name'], result)
+            if user_avatar_url:
+                # Verify if there is a checkbox for the user choose to use their
+                # social avatar.
+                self.assert_in_success_response(['use_social_avatar'], result)
             # Verify the name field gets correctly pre-populated:
             self.assert_in_success_response([expected_final_name], result)
 
-            # Click confirm registration button.
-            result = self.client_post(
-                '/accounts/register/',
-                {'full_name': expected_final_name,
-                 'key': confirmation_key,
-                 'terms': True})
+            # When a user choose not to use their social avatar, the registered url won't be accessed.
+            with responses.RequestsMock(assert_all_requests_are_fired=False) as mock_upload:
```

Comment:
`assert_all_requests_are_fired=True` should be better - it'll throw an error if a request to the mocked url isn't made. Well, if the avatar gets set correctly, as verified by the later block, then that kind of means the request must have been made, so this shouldn't make much difference - but it seems more natural to have `True` than `False`.